src/Runtime/Distributions/Automata/TransducerBase.cs

```diff
@@ -387,7 +388,30 @@ public TDestAutomaton ProjectSource(TSrcAutomaton srcAutomaton)
                             continue;
                         }
 
-                        var destWeight = Weight.Product(mappingTransition.Weight, srcTransition.Weight, Weight.FromLogValue(projectionLogScale));
+                        // In the special case of a log probability override in a DiscreteChar element distribution,
+                        // we need to compensate for the fact that the distribution is not normalized.
+                        if (destElementDistribution.HasValue &&
```

Comment:
Similarly, does it make sense to move this to StringTransducer?

Reply:
Well, the logic is mainly inside the loop in TransducerBase. We could create a virtual method to do the adjustment and override it in StringTransducer, but I am not convinced that this is justified.